nbp/bodies/body_state.py

```diff
@@ -11,6 +11,7 @@ class BodyState(object):
     def __init__(self, bodies: [Body], ticks: int, time: float, delta_time: float):
         self.bodies, self.ticks = bodies, int(ticks)
         self.time, self.delta_time = float(time), float(delta_time)
+        self.time_settings = {'time': [],'radius': []}#All the radi must be ALWAYS be positive and all the time must be either all negative or all positive.
```

Comment:
You should NOT change the BodyState itself. Use the Modificator-pattern for this kind of things and let the Modifier handle state.